packages/open-api-gateway/src/project/smithy/setup-smithy-build.ts

```diff
@@ -41,10 +41,8 @@ export const setupSmithyBuild = (
   options: SmithyApiGatewayProjectOptions
 ): SmithyBuildProjectResult => {
   const modelDir = options.modelDir ?? "model";
-  const fullyQualifiedServiceName =
-    options.serviceName ?? "example.hello#Hello";
-
-  const [serviceNamespace, serviceName] = fullyQualifiedServiceName.split("#");
+  const { namespace: serviceNamespace, serviceName } = options.serviceName;
+  const fullyQualifiedServiceName = `${serviceNamespace}#${serviceName}`;
```

Comment:
Shouldn't we validate these to ensure proper casing and usage of periods for namespaces?

Reply:
I actually started writing validation regexes but then figured it'd end up being validated by the Smithy compiler during smithy build anyway - so rather than repeat the validation a tad earlier I felt like it's simpler to just let Smithy do it :). The only downside with delegating I suppose is Smithy would error _after_ we've generated the model, so the user would need to either delete the model or update it.